statsproxy.go

```diff
@@ -1,206 +1,115 @@
 package main
 
 import (
-	"bytes"
+	_ "expvar"
 	"flag"
 	"fmt"
-	"io"
+	"github.com/fullcontact/statsproxy/common"
+	"github.com/fullcontact/statsproxy/config"
+	"github.com/fullcontact/statsproxy/workers"
 	"log"
 	"net"
+	"net/http"
 	"os"
 	"os/signal"
 	"runtime"
-	"runtime/pprof"
-	"strconv"
-	"strings"
-	"sync/atomic"
 	"syscall"
-	"time"
-)
-
-const (
-	MAX_UDP_PACKET_SIZE = 512
-	SOCKET_READ_BUFFER  = 4194304
-)
-
-var (
-	serviceAddress = flag.String("address", ":8125", "UDP service address")
-	debug          = flag.Bool("debug", false, "print verbose data about packets sent and servers chosen")
-	showVersion    = flag.Bool("version", false, "print version string")
-	workers        = flag.Int("workers", 2, "number of goroutines working on incoming UDP data")
-	hosts          = Hosts{}
-	cpuprofile     = flag.String("cpuprofile", "", "write cpu profile to file")
-	tickerPeriod   = flag.Duration("tickerPeriod", 5*time.Second, "writes internal stats every tickerPeriod seconds")
-
-	// State
-	hostConnections      = make(map[*Host]io.Writer)
-	packetCount          uint32
-	packetCountPerServer = make(map[*Host]*uint32)
 )
 
 var signalchan chan os.Signal
 
-func init() {
-	flag.Var(&hosts, "hosts",
```

Comment:
I'll take on updating the README after a round of changes. For one, I plan to widen the config to YAML. Since YAML is a superset of JSON, they should both work.